core/src/main/java/com/github/phantazmnetwork/core/gui/GuiItem.java

```diff
@@ -0,0 +1,182 @@
+package com.github.phantazmnetwork.core.gui;
+
+import com.github.phantazmnetwork.core.inventory.InventoryObject;
+import net.minestom.server.entity.Player;
+import net.minestom.server.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * An interaction-capable item in a GUI. When present in a dynamic GUI, may be animated (its tick method will be called
+ * to update its {@link ItemStack} if necessary, which will then be retrieved through getStack).
+ */
+public interface GuiItem extends ClickHandler, RemoveHandler, ReplaceHandler, InventoryObject {
+    enum ClickType {
+        /**
+         * Represents a left click.
+         */
+        LEFT_CLICK,
+
+        /**
+         * Represents a right click.
+         */
+        RIGHT_CLICK,
+
+        /**
+         * Represents a shift click.
+         */
+        SHIFT_CLICK,
+
+        /**
+         * Represents a middle click.
+         */
+        MIDDLE_CLICK,
+
+        /**
+         * Represents a double click.
+         */
+        DOUBLE_CLICK
+    }
+
+    /**
+     * Builder for a GuiItem implementation.
+     */
+    class Builder {
+        private final Collection<ClickHandler> clickHandlers;
+        private final Collection<RemoveHandler> removeHandlers;
+        private final Collection<ReplaceHandler> replaceHandlers;
+        private ItemStack itemStack;
+        private ItemUpdater updater;
+
+        private Builder() {
+            this.clickHandlers = new ArrayList<>(1);
+            this.removeHandlers = new ArrayList<>(1);
+            this.replaceHandlers = new ArrayList<>(1);
+            this.itemStack = ItemStack.AIR;
```

Comment:
could AIR default item stack cause issues with inventories (probably would in vanilla but idk about minestom), also generally I think that requiring an ItemStack makes a bit more sense

Reply:
will test if it causes issues

Reply:
it appears that no issues are caused, you can still click the item even though it isn't visible